src/frontend/src/views/Data/components/SamplesView/components/SampleTableModalManager/components/DownloadModal/components/DownloadButton/mapTsvData.ts

```diff
@@ -0,0 +1,50 @@
+// TODO-TR (mlila): consider moving or restructuring this blob
```

Comment:
you can compare this to the existing function in `src/common/components/library/data_subview/index.tsx`